monero/wallet_client.go

```diff
@@ -172,3 +248,233 @@ func (c *walletClient) GetHeight() (uint64, error) {
+		if !info.Mainnet {
+			return fmt.Errorf("monerod endpoint %s is not a mainnet node", endpoint)
+		}
+	case common.Development:
+		if info.NetType != "fakechain" {
+			return fmt.Errorf("monerod endpoint %s should have a network type of \"fakechain\" in dev mode",
+				endpoint)
+		}
+	default:
+		panic("unhandled environment type")
+	}
+	if env != common.Development && info.Offline {
+		return fmt.Errorf("monerod endpoint %s is offline", endpoint)
+	}
+	if !info.Synchronized {
+		return fmt.Errorf("monerod endpoint %s is not synchronised", endpoint)
+	}
+	return nil
+}
+
+// createWalletRPCService starts a monero-wallet-rpc listening on a random port for tests. The json_rpc
+// URL of the started service is returned.
+func createWalletRPCService(conf *WalletClientConf) (string, *os.Process, error) {
+	walletRPCBin := conf.MoneroWalletRPCPath
+	if walletRPCBin == "" {
+		var err error
+		walletRPCBin, err = getMoneroWalletRPCBin()
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	if conf.MonerodHost == "" {
+		conf.MonerodHost = "127.0.0.1"
+	}
+	if conf.MonerodPort == 0 {
+		switch conf.Env {
+		case common.Mainnet, common.Development:
+			conf.MonerodPort = common.DefaultMoneroDaemonMainnetPort
+		case common.Stagenet:
+			conf.MonerodPort = common.DefaultMoneroDaemonStagenetPort
+		default:
+			panic("unhandled environment value")
+		}
+	}
+
+	if err := validateMonerodConfig(conf.Env, conf.MonerodHost, conf.MonerodPort); err != nil {
+		return "", nil, err
+	}
+
+	if conf.LogPath == "" {
+		// default to the folder above the wallet
+		conf.LogPath = path.Join(path.Dir(path.Dir(conf.WalletFilePath)), "monero-wallet-rpc.log")
+	}
+
+	walletRPCBinArgs := getWalletRPCFlags(conf)
+	proc, err := launchMoneroWalletRPCChild(walletRPCBin, walletRPCBinArgs...)
+	if err != nil {
+		return "", nil, fmt.Errorf("%w, see %s for details", err, conf.LogPath)
+	}
+
+	return fmt.Sprintf("http://127.0.0.1:%d/json_rpc", conf.WalletPort), proc, nil
+}
+
+// getMoneroWalletRPCBin returns the monero-wallet-rpc binary. It first looks for
+// "./monero-bin/monero-wallet-rpc". If not found, it then looks for "monero-wallet-rpc"
+// in the user's path.
+func getMoneroWalletRPCBin() (string, error) {
+	execName := "monero-wallet-rpc"
+	priorityPath := path.Join("monero-bin", execName)
+	path, err := exec.LookPath(priorityPath)
+	if err == nil {
+		return path, nil
+	}
+	if !errors.Is(err, os.ErrNotExist) {
+		return "", err
+	}
+	// search for the executable in the user's PATH
+	return exec.LookPath(execName)
+}
+
+func launchMoneroWalletRPCChild(walletRPCBin string, walletRPCBinArgs ...string) (*os.Process, error) {
+	cmd := exec.Command(walletRPCBin, walletRPCBinArgs...)
+
+	pRead, pWrite, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	cmd.Stdout = pWrite
+	cmd.Stderr = pWrite
+
+	// Last entry wins if an environment variable is in the list multiple times.
+	// We parse some output, so we want to force English. NO_COLO[U]R=1 failed to
```

Comment:
This comment line was killing me.  The commonly used environment variable is only `NO_COLOR`, not `NO_COLOUR`: https://no-color.org/

Unfortunately, there is no way to add a linter exception, at least not that I could figure out, in a line that is already a comment, so I had to pretend that the variable might have multiple spellings.  :)

Reply:
you can add a `//nolint:misspell` to the top of the whole comment!

Reply:
Thanks!  That worked.